hikari/impl/special_endpoints.py

```diff
@@ -1527,33 +1525,36 @@ def url(self) -> str:
 
 
 @attr.define(kw_only=True, weakref_slot=False)
-class InteractiveButtonBuilder(
-    _ButtonBuilder[_ContainerProtoT], special_endpoints.InteractiveButtonBuilder[_ContainerProtoT]
-):
+class InteractiveButtonBuilder(_ButtonBuilder, special_endpoints.InteractiveButtonBuilder):
     """Builder class for interactive buttons."""
 
     _custom_id: str = attr.field(alias="custom_id")
+    _url: undefined.UndefinedType = attr.field(init=False, default=undefined.UNDEFINED)
 
     @property
     def custom_id(self) -> str:
         return self._custom_id
 
 
 @attr_extensions.with_copy
-@attr.define(kw_only=True, weakref_slot=False)
-class _SelectOptionBuilder(special_endpoints.SelectOptionBuilder["_TextSelectMenuBuilderT"]):
+@attr.define(weakref_slot=False)
+class SelectOptionBuilder(special_endpoints.SelectOptionBuilder):
     """Builder class for select menu options."""
 
-    _menu: _TextSelectMenuBuilderT = attr.field(alias="menu")
     _label: str = attr.field(alias="label")
     _value: str = attr.field(alias="value")
-    _description: undefined.UndefinedOr[str] = attr.field(alias="description", default=undefined.UNDEFINED)
+    _description: undefined.UndefinedOr[str] = attr.field(
+        alias="description", default=undefined.UNDEFINED, kw_only=True
+    )
     _emoji: typing.Union[snowflakes.Snowflakeish, emojis.Emoji, str, undefined.UndefinedType] = attr.field(
-        alias="emoji", default=undefined.UNDEFINED
+        alias="emoji", default=undefined.UNDEFINED, kw_only=True
     )
-    _emoji_id: undefined.UndefinedOr[str] = attr.field(alias="emoji_id", default=undefined.UNDEFINED)
-    _emoji_name: undefined.UndefinedOr[str] = attr.field(alias="emoji_name", default=undefined.UNDEFINED)
-    _is_default: bool = attr.field(alias="is_default", default=False)
+    _emoji_id: undefined.UndefinedOr[str] = attr.field(init=False, default=undefined.UNDEFINED)
+    _emoji_name: undefined.UndefinedOr[str] = attr.field(init=False, default=undefined.UNDEFINED)
+    _is_default: bool = attr.field(alias="is_default", default=False, kw_only=True)
+
+    def __attrs_post_init__(self) -> None:
+        self._emoji_id, self._emoji_name = _build_emoji(self._emoji)
```

Comment:
Instead of storing this in this way, cant we just keep `self._emoji` and then calculate what it should be in the `build`?

Reply:
These models are designed to be reused, it just does the check once rather than doing it every time its built